python/lib/sift_py/upload/config.py

```diff
@@ -0,0 +1,149 @@
+    format: str
+    column_number: int
+    relative_start_time: Optional[str] = None
+
+    @model_validator(mode="after")
+    def validate_format(self):
+        if self.format not in VALID_TIME_FORMATS:
+            raise PydanticCustomError(
+                "invalid_config_error",
+                "Invalid time format: {format}.\nValid options: {valid}",
+                {"format": self.format, "valid": ", ".join(VALID_TIME_FORMATS)},
+            )
+
+        return self
+
+    @model_validator(mode="after")
+    def validate_relative_time(self):
+        if self.format.startswith("TIME_FORMAT_RELATIVE_"):
+            if self.relative_start_time is None:
+                raise PydanticCustomError("invalid_config_error", "Missing 'relative_start_time'")
+        else:
+            if self.relative_start_time:
+                raise PydanticCustomError(
+                    "invalid_config_error",
+                    "'relative_start_time' specified for non relative time format.",
+                )
+
+        return self
+
+
+class DataColumn(_BaseModel):
+    name: str
+    data_type: str
+    component: Optional[str] = ""
+    units: Optional[str] = ""
+    description: Optional[str] = ""
+    enum_types: Optional[List[EnumType]] = []
+    bit_field_elements: Optional[List[BitFieldElement]] = []
+
+    @model_validator(mode="after")
+    def validate_data_type(self):
+        if self.data_type not in VALID_DATA_TYPES:
+            raise PydanticCustomError(
+                "invalid_config_error",
+                f"Invalid data_type: {self.data_type}.\nValid options: {', '.join(VALID_DATA_TYPES)}",
+            )
+
+        return self
+
+    @model_validator(mode="after")
+    def validate_enums(self):
+        if self.enum_types:
+            if self.data_type != "CHANNEL_DATA_TYPE_ENUM":
+                raise PydanticCustomError(
+                    "invalid_config_error",
+                    f"Enums can only be specified with the CHANNEL_DATA_TYPE_ENUM data type. {self.name} is {self.data_type}",
+                )
+
+        return self
+
+    @model_validator(mode="after")
+    def validate_bit_fields(self):
+        if self.bit_field_elements:
+            if self.data_type != "CHANNEL_DATA_TYPE_BIT_FIELD":
+                raise PydanticCustomError(
+                    "invalid_config_error",
+                    f"Bit fields can only be specified with the CHANNEL_DATA_TYPE_BIT_FIELD data type. {self.name} is {self.data_type}",
+                )
+
+        return self
+
+
+class _CsvConfigImpl(_BaseModel):
+    """ """
+
+    asset_name: str
+    run_name: Optional[str] = None
+    run_id: Optional[str] = None
+    first_data_row: int
+    time_column: TimeColumn
+    data_columns: Dict[int, DataColumn]
+
+    @model_validator(mode="after")
+    def validate_config(self):
+        if not self.data_columns:
+            raise PydanticCustomError("invalid_config_error", "Empty 'data_columns'")
+
+
+class CsvConfig:
```

Comment:
Could we move this fella to the top of the file? It seems like the primary purpose of this module. If you `from __future__ import annotations` then python will allow it.